code/datums/langchat/langchat.dm

```diff
@@ -151,6 +151,43 @@
 	else
 		addtimer(CALLBACK(src, /atom.proc/langchat_drop_image, language), timer, TIMER_OVERRIDE|TIMER_UNIQUE|TIMER_NO_HASH_WAIT)
 
+//used for broadcasting from a vehicle
+/mob/proc/langchat_vehicle_broadcast(message, var/list/listeners, language, var/obj/vehicle/multitile/V)
+
+	if(!V || interactee != V)
+		return
+
+	langchat_drop_image()
+	langchat_make_image()
+
+	var/text_left = null
+	var/text_to_display = message
+
+	if(length(message) > LANGCHAT_LONGEST_TEXT)
+		text_to_display = copytext_char(message, 1, LANGCHAT_LONGEST_TEXT - 5) + "..."
+		text_left = "..." + copytext_char(message, LANGCHAT_LONGEST_TEXT - 5)
+	var/timer = 6 SECONDS
+	if(text_left)
+		timer = 4 SECONDS
+	text_to_display = "<span class='center [langchat_styles] langchat_announce langchat'>[text_to_display]</span>"
+
+	langchat_image.maptext = text_to_display
+	langchat_image.maptext_width = LANGCHAT_WIDTH * 2
+
+	langchat_listeners = listeners
+	for(var/mob/M in langchat_listeners)
```

Comment:
better var name needed for M

Reply:
1. Why?
2. It's how it is in other two procs.